### PR TITLE
New Rule: disallowNestedTernaries

### LINKED ISSUE #1824: New Rule: disallowNestedTernaries
disallowNestedTernaries/disallowNestedConditionalExpressions

Don't know if we want to make a parameter for a certain maxLevel of nesting since that might be too much right now?

Ref https://github.com/jscs-dev/node-jscs/pull/1806#issuecomment-142786850
@bjdixon

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27069749-new-rule-disallownestedternaries?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by bjdixon**
Fixes #1824

### CODE REVIEW

**Comment by hzoo** [comment]
Wow awesome - looks pretty good! Nice idea to have a param for a max level of nesting. I guess you could also change it to maybe skip ConditionalExpressions that have already been visited due to nesting but it's probably not necessary as people don't normally nest that much

**Comment by markelog** [comment]
When we will have CST, that would be even better for autofixing.

@bjdixon would you like us to ping you when it will happen?

File: lib/rules/disallow-nested-ternaries.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    getOptionName: function() {
+        return 'disallowNestedTernaries';
+    },
+
+    check: function(file, errors) {
+        var maxLevel = this._maxLevel;
+        file.iterateNodesByType('ConditionalExpression', function(node) {
+            var level = 0;
+            var getLevel = function(currentNode) {
```
Comment (markelog) [inline lib/rules/disallow-nested-ternaries.js]: Is this helper really needed?

Comment (markelog) [inline lib/rules/disallow-nested-ternaries.js]: Oh, i see, yeah, cool

**Comment by bjdixon** [comment]
@hzoo Maybe I should return a value from ```getLevel``` and cache that? I think that might reduce the redundant walking, but it would introduce a little more complexity. I'm probably missing a simpler strategy that you have in mind. I'm happy to implement this differently. Let me know.

@markelog yeah, definitely ping me for any updates.

**Comment by hzoo** [comment]
Yeah I figured maybe the complexity isn't worth it? Or, maybe it should exit early based on maxLevels so you don't need to check? Not sure

**Comment by bjdixon** [comment]
Agreed, returning early would be better. There's really no point in checking past the boundary. I'll update the PR.

**Comment by hzoo** [comment]
ok nice @markelog?

**Comment by markelog** [comment]
LGTM

**Comment by hzoo** [comment]
Awesome stuff again @bjdixon!

**Pull request closed without merging**